### ISSUE TITLE
False positive for SC2127 on `;&`

### ISSUE THREAD

**Issue by cD1rtX3** [koalaman/shellcheck#3416]
#### For bugs with existing features

- Rule Id: SC2127
- My shellcheck version: online
- [X] The rule's wiki page does not already cover this
- [X] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/sh

case "$1" in
    a);&
esac
```

#### Here's what shellcheck currently says:

> [SC2127](https://www.shellcheck.net/wiki/SC2127) (error): To use cases with ;&, specify #!/usr/bin/env bash or ksh

#### Here's what I wanted or expected to see:

Nothing.

#### Additional comments:

See POSIX [9799919799§2.9.4.3](https://pubs.opengroup.org/onlinepubs/9799919799/utilities/V3_chap02.html#tag_19_09_04_05).

**Comment by e-kwsm** [comment]
duplicate of #3166
